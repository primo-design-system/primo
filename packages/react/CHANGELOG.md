# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.5](https://github.com/primodesignsystem/primo/compare/@primo/react@1.0.4...@primo/react@1.0.5) (2023-01-10)

**Note:** Version bump only for package @primo/react

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.4](https://github.com/primodesignsystem/primo/compare/@primo/react@1.0.2...@primo/react@1.0.4) (2023-01-09)

**Note:** Version bump only for package @primo/react

## @primo/react [1.0.2](https://github.com/primo-design-system/primo/compare/@primo/react@1.0.1...@primo/react@1.0.2) (2022-12-12)

### Dependencies

- **@primo/tokens:** upgraded to 1.0.1
- **@primo/styles:** upgraded to 1.1.0
- **@primo/components:** upgraded to 2.0.0

## @primo/react [1.0.1](https://github.com/primo-design-system/primo/compare/@primo/react@1.0.0...@primo/react@1.0.1) (2022-11-19)

### Dependencies

- **@primo/styles:** upgraded to 1.0.1
- **@primo/components:** upgraded to 1.0.1

# @primo/react 1.0.0 (2022-11-18)

### Features

- new button changes + playrom and new tests
  ([c1da249](https://github.com/primo-design-system/primo/commit/c1da24929b594f069e18bc40003041e37f4b6811))
- updating docs and tests
  ([0f95c7c](https://github.com/primo-design-system/primo/commit/0f95c7ca2bd271f25333f017d7b5e770d4c43462))
- **component:** build scripts for button component and linting
  ([ba76456](https://github.com/primo-design-system/primo/commit/ba76456f523a53d09c0605615ebf9f7751f129a7))
- **react:** new ButtonGroup component, setup and tests
  ([8685ea5](https://github.com/primo-design-system/primo/commit/8685ea55fd7d6dba68eb72240bdbcaece1bdaf43))
- **tokens:** including new token values
  ([10f1161](https://github.com/primo-design-system/primo/commit/10f11615e87e00bcc691c18ccd04913c1bec8362))
- **tokens:** new colour and font tokens
  ([38efa21](https://github.com/primo-design-system/primo/commit/38efa21d6e0c487f10e5aacf6f0a030d3170dcf1))

### Dependencies

- **@primo/tokens:** upgraded to 1.0.0
- **@primo/styles:** upgraded to 1.0.0
- **@primo/components:** upgraded to 1.0.0
