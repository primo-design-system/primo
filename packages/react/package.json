{
  "name": "@primo/react",
  "version": "1.0.2",
  "description": "> TODO: description",
  "author": "Richard McCartney <richard@mccartney.io>",
  "homepage": "https://github.com/primodesignsystem/primo#readme",
  "license": "MIT",
  "main": "dist/index.js",
  "module": "dist/index.es.js",
  "style": "dist/index.css",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c",
    "build:playroom": "playroom build",
    "build:storybook": "build-storybook --docs --quiet",
    "build:storybook:quiet": "yarn storybook --quiet",
    "build:watch": "rollup -c -w",
    "lint": "yarn lint:es && yarn lint:ts",
    "lint:es": "eslint src/**/*.ts[x]",
    "lint:es:fix": "yarn lint:es --fix",
    "lint:ts": "tslint -c tslint.json 'src/**/*.{ts,tsx}'",
    "lint:ts:fix": "yarn lint:ts --fix",
    "start:playroom": "playroom start",
    "start:storybook": "start-storybook -p 6006 -s public",
    "test": "jest",
    "test:update-snapshots": "yarn test -u"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/primodesignsystem/primo.git"
  },
  "bugs": {
    "url": "https://github.com/primodesignsystem/primo/issues"
  },
  "devDependencies": {
    "@babel/core": "^7.11.5",
    "@primo/tokens": "1.0.1",
    "@primo/styles": "1.1.0",
    "@primo/components": "2.0.0",
    "@rollup/plugin-commonjs": "^23.0.2",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "@rollup/plugin-typescript": "^9.0.2",
    "@storybook/addon-a11y": "^6.2.9",
    "@storybook/addon-essentials": "^6.2.9",
    "@storybook/addon-links": "^6.2.9",
    "@storybook/addon-postcss": "^2.0.0",
    "@storybook/react": "^6.5.12",
    "@storybook/theming": "^6.2.9",
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.0.4",
    "@types/classnames": "^2.2.10",
    "@types/jest": "^26.0.14",
    "@types/jest-axe": "^3.5.5",
    "@types/node": "^12.12.38",
    "@types/prop-types": "^15.7.3",
    "@types/react": "^16.9.49",
    "@types/react-dom": "^16.9.8",
    "@types/storybook__react": "^5.2.1",
    "@types/testing-library__jest-dom": "^5.9.2",
    "@types/testing-library__react": "^10.2.0",
    "@typescript-eslint/eslint-plugin": "^4.0.0",
    "@typescript-eslint/parser": "^4.0.0",
    "@whitespace/storybook-addon-html": "^5.0.0",
    "babel-eslint": "^10.0.0",
    "babel-loader": "^8.1.0",
    "cheerio": "^1.0.0-rc.3",
    "css-loader": "^6.7.1",
    "file-loader": "6.2.0",
    "jest": "^26.4.2",
    "jest-axe": "^6.0.0",
    "playroom": "0.28.0",
    "postcss": "^8.2.4",
    "postcss-import": "^12.0.1",
    "postcss-preset-env": "^6.7.0",
    "prettier": "^1.19.1",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-is": "^16.13.1",
    "rollup": "^2.26.9",
    "rollup-plugin-peer-deps-external": "^2.2.3",
    "rollup-plugin-postcss": "^3.1.8",
    "style-loader": "^3.3.1",
    "ts-jest": "^26.4.0",
    "tslib": "^2.0.1",
    "tslint": "^6.1.3",
    "typescript": "^4.0.2",
    "url-loader": "^4.1.1",
    "webpack": "^5.2.0"
  }
}
