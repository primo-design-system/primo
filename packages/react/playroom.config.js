const exampleCode = `
  <Button>Hello World</Button>
`;

module.exports = {
  components: './src/components/index.ts',
  outputPath: './dist/playroom',

  // Optional:
  title: 'Primo Design System',
  widths: [320, 768, 1024],
  port: 9000,
  openBrowser: true,
  paramType: 'search', // default is 'hash'
  frameComponent: './.playroom/FrameComponent.tsx',
  exampleCode: exampleCode,
  baseUrl: '/playroom/',
  webpackConfig: () => ({
    module: {
      rules: [
        {
          test: /\.tsx?$/,
          include: __dirname,
          use: {
            loader: 'babel-loader',
            options: {
              presets: ['@babel/preset-env', '@babel/preset-typescript', '@babel/preset-react'],
            },
          },
        },
        {
          test: /\.css$/,
          exclude: /node_modules\/(?!(@primo)\/).*/,
          use: [require.resolve('style-loader'), require.resolve('css-loader')],
        },
      ],
    },
    resolve: {
      extensions: ['.js', '.ts', '.tsx', '.css', '.pcss'],
      modules: ['node_modules'],
    },
  }),
  iframeSandbox: 'allow-scripts',
};
