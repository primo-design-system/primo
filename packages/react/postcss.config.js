module.exports = {
  extract: true,
  minimize: true,
  plugins: [require('postcss-import'), require('postcss-preset-env')],
};
