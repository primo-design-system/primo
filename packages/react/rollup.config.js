import typescript from '@rollup/plugin-typescript';
import commonjs from '@rollup/plugin-commonjs';
import resolve from '@rollup/plugin-node-resolve';
import external from 'rollup-plugin-peer-deps-external';
import postcss from 'rollup-plugin-postcss';

const basePlugins = [external(), resolve(), commonjs()];
const postcssConfig = require('./postcss.config');

/**
 * We have multiple output configs to support CommonJS and ES modules.
 */
export default [
  // CommonJS
  {
    input: 'src/index.ts',
    output: {
      dir: 'dist',
      format: 'cjs',
      exports: 'named',
      sourcemap: true,
    },

    plugins: [...basePlugins, postcss(postcssConfig), typescript({ declaration: true, declarationDir: 'dist/types' })],
  },
  // ES Module
  {
    input: 'src/index.ts',
    output: [
      {
        file: 'dist/index.es.js',
        format: 'es',
        exports: 'named',
        sourcemap: true,
      },
    ],
    plugins: [
      ...basePlugins,
      postcss({
        ...postcssConfig,
        extract: false,
        inject: false,
      }),
      typescript(),
    ],
  },
];
