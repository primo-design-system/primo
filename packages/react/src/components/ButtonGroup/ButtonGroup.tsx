import React, {
  FunctionComponent,
  ReactNode,
  forwardRef
} from 'react';

type RefType = React.RefObject<HTMLDivElement>;

export interface ButtonGroupProps {
  /** Specify the content of your ButtonGroup */
  children?: ReactNode;

  /** Optional prop to specify the ID of the ButtonGroup */
  id?: string;

  /** Optional prop to specify the ID used for testing */
  testId?: string;
}

/**
 * ButtonGroup component.
 *
 * @component
 * @example
 * return (
 *   <ButtonGroup />
 * )
 */
const ButtonGroup: FunctionComponent<ButtonGroupProps> = forwardRef(function ButtonGroup(
  {
    children,
    id,
    testId,
    ...props
  },
  ref: RefType
) {
  return (
    <div id={id} ref={ref} data-testid={testId} {...props}>
      {children}
    </div>
  );
});

export default ButtonGroup;
