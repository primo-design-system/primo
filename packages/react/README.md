# @primo/react

[![NPM version](https://badgen.net/npm/v/@primo/react)](https://www.npmjs.org/package/@primo/react)


> The Primo Design System components built with [React](https://reactjs.org/)

## Getting started

To install `@primo/react` to your new or exsisting project, you will
need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install -S @primo/react
```

If you use [Yarn](https://yarnpkg.com/en/), use the following command instead:

```bash
yarn add @primo/react
```

This package requires [React](https://www.npmjs.com/package/react) in order to
compile.

## Usage

```js
import { Button } from '@primo/react';

function HelloWorld() {
  return <Button>Example button</Button>;
}
```

## ‚úçÔ∏è Feedback

Please ask any questions on this project, you can do so by
[creating a new issue](https://github.com/primo-design-system/primo/issues/new/choose). Or contributing to any [active issues](https://github.com/primo-design-system/primo/issues).

If you wish to get in contact, please don't hesitate via twitter [@Rich_McCartney](https://twitter.com/rich_mccartney).

## üìù Licence

Licensed under standard
[MIT license](https://github.com/primo-design-system/primo/blob/main/LICENSE).
