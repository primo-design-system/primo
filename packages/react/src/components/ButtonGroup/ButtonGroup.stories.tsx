import { Meta, Story } from "@storybook/react";
import React from "react";

import ButtonGroup, { ButtonGroupProps } from "./ButtonGroup";

export default {
  args: { children: "ButtonGroup" },
  argTypes: { className: { control: { type: null } } },
  component: ButtonGroup,
  parameters: {
    edgeToEdge: false,
    withSurface: false
  },
  title: "Components/ButtonGroup"
} as Meta;

export const Default: Story<ButtonGroupProps> = args => <ButtonGroup {...args} />;
