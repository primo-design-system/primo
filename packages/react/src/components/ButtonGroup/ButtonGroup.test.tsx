import React from 'react';

import { render, cleanup, screen } from "@testing-library/react";
import ButtonGroup from '.';

describe('<ButtonGroup />', () => {
  afterEach(cleanup);

  const defaultProps = {
    testId: 'buttonGroup'
  };

  test(": matches snapshot", () => {
    const { container } = render(<ButtonGroup {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  test(": renders", () => {
    render(<ButtonGroup {...defaultProps}>Click me</ButtonGroup>);
    const buttonGroup = screen.getByTestId('buttonGroup');

    expect(buttonGroup).toBeInTheDocument();
  });
});
