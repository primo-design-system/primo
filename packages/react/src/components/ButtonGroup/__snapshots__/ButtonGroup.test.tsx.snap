// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> : matches snapshot 1`] = `
<div>
  <div
    data-testid="buttonGroup"
  />
</div>
`;
